File: java/src/main/java/axon/AxonType.java
package axon;

public enum AxonType {
    STRING('S'),
    INTEGER('I'),
    FLOAT('F'),
    BOOLEAN('B'),
    TIMESTAMP('T');
    
    private final char code;
    
    AxonType(char code) {
        this.code = code;
    }
    
    public char getCode() {
        return code;
    }
    
    public static AxonType fromCode(char code) {
        for (AxonType type : values()) {
            if (type.code == code) {
                return type;
            }
        }
        throw new IllegalArgumentException("Unknown type code: " + code);
    }
}

File: csharp/src/Axon/AxonType.cs
namespace Axon;

/// <summary>
/// Represents the data types supported by the AXON format.
/// </summary>
public enum AxonType
{
    /// <summary>
    /// String type (S).
    /// </summary>
    String,

    /// <summary>
    /// Integer type (I).
    /// </summary>
    Integer,

    /// <summary>
    /// Floating-point type (F).
    /// </summary>
    Float,

    /// <summary>
    /// Boolean type (B).
    /// </summary>
    Boolean,

    /// <summary>
    /// Timestamp type (T).
    /// </summary>
    Timestamp,

    /// <summary>
    /// Decimal type (D) - precise decimal numbers without floating-point artifacts.
    /// </summary>
    Decimal,
}
